### PR TITLE
Azure config starter

### PR DESCRIPTION
**Pull request by sophiaso**
## Description
Add Spring Cloud Azure Config Starter

### CODE REVIEW

File: spring-cloud-azure-samples/spring-cloud-azure-config-sample/pom.xml [hunk 1]
```diff
@@ -14,7 +14,7 @@
     <dependencies>
         <dependency>
             <groupId>com.microsoft.azure</groupId>
-            <artifactId>spring-cloud-azure-config</artifactId>
+            <artifactId>spring-cloud-starter-azure-config</artifactId>
```
Comment (warrenzhu25) [inline spring-cloud-azure-samples/spring-cloud-azure-config-sample/pom.xml]: With this, we should not need `spring-boot-starter`

File: spring-cloud-azure-samples/spring-cloud-azure-config-sample/src/main/resources/bootstrap.properties [hunk 2]
```diff
@@ -1,4 +1,3 @@
 spring.cloud.azure.config.connection-string=put-your-connection-string-here
```
Comment (warrenzhu25) [inline spring-cloud-azure-samples/spring-cloud-azure-config-sample/src/main/resources/bootstrap.properties]: I just want to confirm whether we really need to rely on connections string. As all other spring cloud modules don't have such low level dependency, they only rely on logical concepts. And our design goal is to minimize such low level details.

File: spring-cloud-azure-samples/spring-cloud-azure-config-sample/pom.xml (hunk 1 above)
Comment (sophiaso) [reply spring-cloud-azure-samples/spring-cloud-azure-config-sample/pom.xml]: Updated.

File: spring-cloud-azure-samples/spring-cloud-azure-config-sample/src/main/resources/bootstrap.properties (hunk 2 above)
Comment (sophiaso) [reply spring-cloud-azure-samples/spring-cloud-azure-config-sample/src/main/resources/bootstrap.properties]: Will improve if possible in the future.

**Pull request merged**